File: src/lib/age.ts
export function parseAgeDays(val: string): number {
  const m = val.trim().match(/^(\d+)(d|w|m)?$/i);
  if (!m) throw new Error(`Invalid --older-than value: '${val}'. Use a number of days (e.g. 30, 30d, 4w, 1m).`);
  const n = parseInt(m[1], 10);
  const unit = (m[2] ?? "d").toLowerCase();
  if (unit === "w") return n * 7;
  if (unit === "m") return n * 30;
  return n;
}

File: src/util/age.rs
use anyhow::{bail, Result};

pub fn parse_age_days(val: &str) -> Result<u64> {
    let val = val.trim();
    let (digits, unit) = val.split_at(
        val.find(|c: char| c.is_alphabetic()).unwrap_or(val.len()),
    );
    let n: u64 = digits.parse().map_err(|_| anyhow::anyhow!("invalid number in '{val}'"))?;
    match unit.to_lowercase().as_str() {
        "" | "d" => Ok(n),
        "w" => Ok(n * 7),
        "m" => Ok(n * 30),
        other => bail!("invalid unit '{other}' in '{val}'. Use d, w, or m"),
    }
}
